### PR TITLE
RDM-10041 Create CaseLocation complex type

### PR DESCRIPTION
**Pull request by Abi-Laver**
https://tools.hmcts.net/jira/browse/RDM-10041

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[X] No
```

### CODE REVIEW

File: repository/src/main/resources/db/migration/V0002__RDM-10041.sql [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+    (select id from field_type where reference = 'Complex'
+        and jurisdiction_id is null
+        and version = (select max(version)
+        from field_type where reference = 'Complex'
+        and jurisdiction_id is null
+        and base_field_type_id is null))
+);
+
+insert into complex_field (reference, label, security_classification, field_type_id, complex_field_type_id)
+values ('RegionId', 'RegionId', 'PUBLIC',
```
Comment (mario-paniccia) [inline repository/src/main/resources/db/migration/V0002__RDM-10041.sql:12]: can you ask the BA what is the label they would want to appear on the UI for these 2 subfields? 
Not sure whether `RegionId` is a very human readable label. Wouldn't `Region id` be better? same for the other field

File: repository/src/main/resources/db/migration/V20201218__RDM-10041__CaseLocation.sql [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+    (select id from field_type where reference = null
+        and jurisdiction_id is null
+        and version = (select max(version)
+        from field_type where reference = null
+        and jurisdiction_id is null
+        and base_field_type_id is null))
+);
+
+insert into complex_field (reference, label, security_classification, field_type_id, complex_field_type_id)
+values ('Region ID', 'Region ID', 'PUBLIC',
```
Comment (mario-paniccia) [inline repository/src/main/resources/db/migration/V20201218__RDM-10041__CaseLocation.sql:31]: according to the ticket this complex subfield id should be `RegionId`

File: repository/src/main/resources/db/migration/V20201218__RDM-10041__CaseLocation.sql [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+    (select id from field_type where reference = 'Region' and version = 1 and jurisdiction_id is null),
+    (select id from field_type where reference = 'CaseLocation' and version = 1 and jurisdiction_id is null));
+
+insert into complex_field (reference, label, security_classification, field_type_id, complex_field_type_id)
+values ('Base Location ID', 'Base Location ID', 'PUBLIC',
```
Comment (mario-paniccia) [inline repository/src/main/resources/db/migration/V20201218__RDM-10041__CaseLocation.sql:36]: according to the ticket this complex subfield id should be `BaseLocationId`
```suggestion
values ('BaseLocationId', 'Base Location ID', 'PUBLIC',
```

File: repository/src/main/resources/db/migration/V20201218__RDM-10041__CaseLocation.sql (hunk 2 above)
Comment (Abi-Laver) [reply repository/src/main/resources/db/migration/V20201218__RDM-10041__CaseLocation.sql:31]: Implemented change

File: repository/src/main/resources/db/migration/V20201218__RDM-10041__CaseLocation.sql [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+
+insert into field_type (created_at, reference, version, base_field_type_id)
+values (now(), 'Region', 1,
+    (select id from field_type where reference = null
+        and jurisdiction_id is null
+        and version = (select max(version)
+        from field_type where reference = null
+        and jurisdiction_id is null
+        and base_field_type_id is null))
+);
```
Comment (mario-paniccia) [inline repository/src/main/resources/db/migration/V20201218__RDM-10041__CaseLocation.sql:19]: I'm sure you copied this so not your fault but I'm not sure I understand what this is doing. 
Consider that FIELD_TYPE.REFERENCE is a non null column, so I don't understand what's the point of even having a condition 
`where reference = null`

Comment (mario-paniccia) [reply repository/src/main/resources/db/migration/V20201218__RDM-10041__CaseLocation.sql:19]: I think we can just leave out setting base_field_type_id which will leave it as NULL, which is what we want for base types

File: repository/src/main/resources/db/migration/V20201218__RDM-10041__CaseLocation.sql [hunk 5]
```diff
@@ -0,0 +1,38 @@
...
+insert into field_type (created_at, reference, version, base_field_type_id)
+values (now(), 'BaseLocation', 1,
+    (select id from field_type where reference = null
+        and jurisdiction_id is null
+        and version = (select max(version)
+        from field_type where reference = null
+        and jurisdiction_id is null
+        and base_field_type_id is null))
+);
```
Comment (mario-paniccia) [inline repository/src/main/resources/db/migration/V20201218__RDM-10041__CaseLocation.sql:28]: same as for `Region`

File: repository/src/main/resources/db/migration/V20201218__RDM-10041__CaseLocation.sql [hunk 6]
```diff
@@ -0,0 +1,38 @@
...
+values (now(), 'CaseLocation', 1,
+    (select id from field_type where reference = 'Complex'
+        and jurisdiction_id is null
+        and version = (select max(version)
+        from field_type where reference = 'Complex'
+        and jurisdiction_id is null
+        and base_field_type_id is null))
+);
+
+insert into field_type (created_at, reference, version, base_field_type_id)
```
Comment (mario-paniccia) [inline repository/src/main/resources/db/migration/V20201218__RDM-10041__CaseLocation.sql:11]: we need to add the configuration for ElasticSearch for these 2 new base types

**Review by mario-paniccia — APPROVED** [review]
👍

**Pull request merged**